src/Integration.UnitTests/Suppression/SuppressionIssueMatcherTests.cs

```diff
@@ -0,0 +1,145 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Integration.Suppression;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.Suppression
+{
+    [TestClass]
+    public class SuppressionIssueMatcherTests
+    {
+        private SuppressionIssueMatcher testSubject;
+        private Mock<ISonarQubeIssuesProvider> mockIssuesProvider;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            mockIssuesProvider = new Mock<ISonarQubeIssuesProvider>();
+            testSubject = new SuppressionIssueMatcher(mockIssuesProvider.Object);
+        }
+
+        [TestMethod]
+        public void Ctor_NullProvider_Throws()
+        {
+            Action act = () => new SuppressionIssueMatcher(null);
+            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("issuesProvider");
+        }
+
+        [TestMethod]
+        public void MatchExists_NullIssue_Throws()
+        {
+            Action act = () => testSubject.SuppressionExists(null);
+            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("issue");
+        }
+
+        [TestMethod]
+        public void MatchExists_NoServerIssues_ReturnsFalse()
+        {
+            // Arrange
+            var issueToMatch = CreateIssueToMatch("rule1", 1, "hash1");
+            ConfigureServerIssues(issueToMatch /* no server issues */);
+
+            // Act and assert
+            testSubject.SuppressionExists(issueToMatch).Should().BeFalse();
+        }
+
+        [TestMethod]
+        public void MatchExists_NoMatches_ReturnsFalse()
+        {
+            // Arrange
+            var issueToMatch = CreateIssueToMatch("RightRuleId", 1, "RightHash");
+
+            ConfigureServerIssues(issueToMatch,
+                CreateServerIssue("WrongRuleId", 1, "RightHash"),  // wrong rule id
+                CreateServerIssue("RightRuleId", 2, "wrong hash"), // wrong line and hash
```

Comment:
Does the line make a difference? 
Maybe change it to be the same line, to emphasize that the hash is what's wrong.